pkg/steps/template_test.go

```diff
@@ -238,3 +242,114 @@ func TestOperateOnTemplatePods(t *testing.T) {
 		})
 	}
 }
+
+func TestInjectLabelsToTemplate(t *testing.T) {
+	testCases := []struct {
+		testID   string
+		jobSpec  *api.JobSpec
+		template *templateapi.Template
+		expected *templateapi.Template
+	}{
+		{
+			testID: "nil refs in jobspec, no injection expected",
+			jobSpec: &api.JobSpec{
+				JobSpec: downwardapi.JobSpec{
+					Refs: nil,
+				},
+			},
+			template: &templateapi.Template{
+				TypeMeta:   meta.TypeMeta{Kind: "Template", APIVersion: "template.openshift.io/v1"},
+				ObjectMeta: meta.ObjectMeta{Name: "test-template"},
+				Objects: []runtime.RawExtension{
```

Comment:
This field is not germane to this test. we should remove it

Reply:
addressed